phac-nml/irida-next
# Public activity: Groups

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in an `Activity` page to a **group** which will list the following activities:

| Type  | Action |
| ------------- | ------------- |
| Group  | Created, Updated, Transferred (in as subgroup, out), Subgroup Removed  |
| Member  | Added, Updated, Removed, Leaves Project  |
| Shared Group  | Shared, Unshared  |
| Project  | Created, Removed, Transferred (in and out)  |

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/user-attachments/assets/52a08c9a-f7e0-41fb-854c-24cca3335635)

![image](https://github.com/user-attachments/assets/a9628784-9f48-46ed-845d-8a702d242362)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Create a `group` and then edit it's attributes (eg. `description`)
2. View the `activity` page for the `group` and there should be 2 entries, one for creating of the group and another for group attributes being updated
3. From the `members` page, Add a couple of members, one as an owner, and another with any access level, and then update the second user (the one with `any` access level`) to another access level
4. View the `activity` page for the project and you should see 3 entries, two for adding new users, and one for updating a members attributes
5. Now remove the user above, then also leave the group with the current user. Keep note of the remaining user who has an owner access level and then sign in with them
6. View the group's `activity` page and you should see entries for a user being removed, and a user removed themselves from the group
7. Share the `group` via the members page to any other group, then `unlink` the shared `group`
8. View the `activity` page and you should see two entries, one for sharing and one for unsharing the group
9. Create a `project` in the `group` and view the `activity` page. There should be an entry for the project creation in the `group` with a clickable link to the `project` in the activity entry
10. Now remove this `project` and go back to the `activity` page. Verify that there is an entry for removal of the `project`. Also verify that the `PUID` is not clickable in the removal or the creation entries for the `project`
11. Create another `group` and within it create a `project`. Transfer this `project` to the `group` in step 1
12. View the `activity` for the `group` and there should be an entry for the `project` being transferred into the `group`
13. Now transfer the above `project` to another `group` and view the `activity` page for the `group` from step 1. There should be an additional entry for the `project` being transferred to another namespace
14. Create 2 new groups, and 1 new subgroup under the second group that was created
15. Transfer the second group into the namespace for the first group from step 14
16. View the activity page for the first group in step 14 and there should be an entry about the second group being transferred in, and view the activity page for the second group which should also  have an entry for the group being transferred under another namespace
17. Delete the subgroup from step 14 and then view the activity for the second group from step 14
18. View the activity page and verify that there is an entry for the subgroup that was removed



## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
